src/EnemyBehaviourEC.cpp

```diff
@@ -160,3 +234,24 @@ void EnemyBehaviourEC::setAggroDistance(float _aggroDistance) {
 void EnemyBehaviourEC::setWithinRange(bool _withinRange) {
     withinRange = _withinRange;
 }
+
+void EnemyBehaviourEC::setSeparationRadius(bool radius) {
+    separationRadius = radius;
+}
+
+void EnemyBehaviourEC::registerInOtherTransforms(TransformComponent* trans) {
+    otherTransform.push_back(trans);
+}
+
+void EnemyBehaviourEC::unRegisterInOtherTransforms(TransformComponent* trans) {
+    auto it = otherTransform.begin();
+    bool found = false;
+    while (it != otherTransform.end() && !found) {
+        if ((*it) == trans)
+            found = true;
+        else
+            it++;
+    }
+    if ((*it) == trans)
```

Comment:
`found` is true if the last iteration evaluated `(*it) == trans` as true. Use this instead of de-referencing a pointer for second time.

```suggestion
    if (found)
```